src/Altinn.App.Core/Features/Correspondence/Models/SendCorrespondencePayload.cs

```diff
@@ -0,0 +1,59 @@
+using Altinn.App.Core.Features.Correspondence.Builder;
+using Altinn.App.Core.Features.Maskinporten;
+using Altinn.App.Core.Models;
+
+namespace Altinn.App.Core.Features.Correspondence.Models;
+
+/// <summary>
+/// Represents the payload for sending a correspondence
+/// </summary>
+public sealed record SendCorrespondencePayload
+{
+    /// <summary>
+    /// The correspondence request to send
+    /// </summary>
+    public required CorrespondenceRequest CorrespondenceRequest { get; init; }
+
+    /// <summary>
+    /// Access token factory delegate (e.g. <see cref="MaskinportenClient.GetAltinnExchangedToken"/>)
+    /// </summary>
+    public required Func<Task<AccessToken>> AccessTokenFactory { get; init; }
+
+    /// <summary>
+    /// The payload contains a <see cref="Models.CorrespondenceRequest"/> instance
+    /// </summary>
+    /// <param name="correspondenceRequest">The correspondence request to send</param>
+    /// <param name="accessTokenFactory">Access token factory delegate (e.g. <see cref="MaskinportenClient.GetAltinnExchangedToken"/>)</param>
+    public static SendCorrespondencePayload WithRequest(
+        CorrespondenceRequest correspondenceRequest,
+        Func<Task<AccessToken>> accessTokenFactory
+    )
+    {
+        return new SendCorrespondencePayload
+        {
+            CorrespondenceRequest = correspondenceRequest,
+            AccessTokenFactory = accessTokenFactory
+        };
+    }
+
+    // TODO: `WithRequest` overloads for built-in authz schemes here
+
+    /// <summary>
+    /// The payload contains an <see cref="ICorrespondenceRequestBuilder"/> builder instance
+    /// </summary>
+    /// <param name="correspondenceRequestBuilder">The correspondence request builder to send</param>
+    /// <param name="accessTokenFactory">Access token factory delegate (e.g. <see cref="MaskinportenClient.GetAltinnExchangedToken"/>)</param>
+    public static SendCorrespondencePayload WithBuilder(
```

Comment:
`WithRequest` and `WithBuilder` in a simple class like this feels kind of verbose to me. Maybe it could be just `static SendCorrespondencePayload Create(CorrespondenceRequest correspondenceRequest)` (with an additional overload for the builder, same name) and `SendCorrespondencePayload WithCustomAuth(...)` or something?

Reply:
`Create()` with overloads make sense, I like it! 

I originally had that union-like pattern with nested classes, so I probably got a bit blind on the naming after abandoning that approach.

As for how to most elegantly deal with the myriad of auth sources, perhaps we should discuss that a bit further?

Using a delegate like I'm doing now makes sense in a general way, but perhaps turns this into an `Create(...)` overload-hell if we want to start auto-magically supporting lots more options.

And I'm not sure if we actually need to support anything besides a Bearer scheme, but we've toyed with the idea of allowing callers to configure the `HttpRequestMessage` directly.

I like the strongly typed `AccessToken` implementation, but I'm not entirely sure it covers us for all the foreseeable scenarios?